packages/ddp-server/livedata_server_tests.js

```diff
@@ -176,9 +176,28 @@ Meteor.publish("livedata_server_test_sub_context", function (connectionId, userI
   var callback = onSubscription[connectionId];
   var methodInvocation = DDP._CurrentMethodInvocation.get();
   var publicationInvocation = DDP._CurrentPublicationInvocation.get();
+
+  // Check the publish function's environment variables and context.
   if (callback) {
     callback.call(this, methodInvocation, publicationInvocation);
   }
+
+  // Check that onStop callback is have the same context as the publish function
+  // and that it runs with the same environment variables as this publis
+  // function.
+  this.onStop(function () {
+    var onStopMethodInvocation = DDP._CurrentMethodInvocation.get();
+    var onStopPublicationInvocation = DDP._CurrentPublicationInvocation.get();
+    callback.call(this, onStopMethodInvocation, onStopPublicationInvocation);
+  });
+
+  // Check that onStop arrow callback runs with the same environment variables
```

Comment:
I mean the arrow function is safer. If it works without, it will also work with.